### PR TITLE
✨Feat(souce-sftp-bulk): file transfer changes

### PR DESCRIPTION
**Pull request by aldogonzalez8**
## What

To be competitive with very large data volumes, we have no other option than to add file-based bulk data transfers.

Changes for file-transfer project. It depends on [this CDK PR](https://github.com/airbytehq/airbyte/pull/47686).

## How
Through the CDK, we will leverage the file-based connector's ability to write a file locally instead of sending records and emit a new record message with metadata related to the file.

```
{
  "type": "RECORD",
  "record": {
    "stream": "simpsons_locations",
    "file": {
      "file_url": "/tmp/airbyte-file-transfer/sftp-testing-for-file-transfer/sftp-folder/simpsons_locations.csv",
      "bytes": 182776,
      "file_relative_path": "sftp-testing-for-file-transfer/sftp-folder/simpsons_locations.csv",
      "modified": 1728330963000,
      "source_file_url": "//sftp-testing-for-file-transfer/sftp-folder/simpsons_locations.csv"
    },
    "emitted_at": 1730226374632,
    "data": {}
  }
}
```

image

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/spec.py [hunk 1]
```diff
@@ -48,6 +48,17 @@ class Config:
         default="/",
         pattern_descriptor="/folder_to_sync",
     )
+    use_file_transfer: bool = Field(title="File Sync (Experimental)", description="Enable file-based bulk load", default=False)
+
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/spec.py:52]: @aldogonzalez8 - Should we drop this boolean? Or not yet?

File: airbyte-integrations/connectors/source-sftp-bulk/integration_tests/spec.json [hunk 2]
```diff
@@ -430,6 +430,52 @@
           "required": ["name", "format"]
         }
       },
+      "use_file_transfer": {
+        "title": "Use File Transfer (Experimental)",
+        "description": "Enable file-based bulk load",
+        "default": false,
+        "type": "boolean"
+      },
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/source-sftp-bulk/integration_tests/spec.json:438]: Should we drop this boolean?

**Review by aaronsteers — APPROVED** [review]
Just a couple questions for confirmation. Looks good to go! 🚀

File: airbyte-integrations/connectors/source-sftp-bulk/integration_tests/spec.json (hunk 2 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-sftp-bulk/integration_tests/spec.json:438]: Yes we do

File: airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/spec.py (hunk 1 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/spec.py:52]: Yes, will just remove before merge, probably right no.

Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-sftp-bulk/source_sftp_bulk/spec.py]: Removed now.

File: airbyte-integrations/connectors/source-sftp-bulk/integration_tests/spec.json (hunk 2 above)
Comment (aldogonzalez8) [reply airbyte-integrations/connectors/source-sftp-bulk/integration_tests/spec.json]: Dropped!

**Pull request merged**